codebasin/report.py

```diff
@@ -81,6 +82,73 @@ def divergence(setmap):
     return d / float(npairs)
 
 
+def utilization(setmap: defaultdict[frozenset[str], int]) -> float:
```

Comment:
Why do you need `defaultdict` for this annotation? Why doesn't `dict[frozenset[str], int]` work?

Reply:
Good point.

I think in this instance it could indeed be `dict`, because we only iterate through the platforms we find in the `setmap`.  But there are other instances of functions accepting a `setmap` where it matters that it's a `defaultdict`, because the code assumes it's safe to execute `setmap[some_set_of_platforms]` and get 0.  I've been trying to use `defaultdict` everywhere to be consistent.

What do you think is better in the short-term: trying to describe what the function would accept without breaking, or trying to use a consistent type-hint for `setmap`?  (Both seem like bad options to me, which is why I recently opened #135 😆.)

Reply:
In that case, would `typing.Mapping` make more sense?

Reply:
`typing.Mapping` wouldn't guarantee that it returns a zero, though, right?  Are you suggesting we use that wherever the function happens to work for non-`defaultdict` setmaps, but still use `defaultdict` where the function strictly requires it?